### PR TITLE
Updating list_metrics() to Iterator pattern.

### PR DESCRIPTION
**Pull request by dhermes**
~~**NOTE**: Has #2651 as diffbase. Unit test changes will be forthcoming.~~

### CODE REVIEW

**Comment by dhermes** [comment]
@daspecster @tseaver PTAL

**Comment by tseaver** [comment]
pyilnt failure:

logging/google/cloud/logging/_gax.py:500:1: E303 too many blank lines (3)

**Comment by dhermes** [comment]
Ahhhh! Darn rebases. PTAL.

Was that your only issue?

**Comment by tseaver** [comment]
I didn't really review it in detail since Travis was red.

**Comment by dhermes** [comment]
OK well it's good to go now. LMK what you think

File: logging/unit_tests/test_client.py [hunk 1]
```diff
@@ -519,22 +539,40 @@ def test_list_metrics_with_paging(self):
...
         }]
-        client = self._makeOne(project=PROJECT, credentials=_Credentials())
-        api = client._metrics_api = _DummyMetricsAPI()
-        api._list_metrics_response = METRICS, None
+        client = self._makeOne(project=PROJECT, credentials=_Credentials(),
+                               use_gax=False)
+        returned = {
+            'metrics': METRICS,
+        }
+        client.connection = _Connection(returned)
```
Comment (tseaver) [inline logging/unit_tests/test_client.py]: *This* version tests the "list all items" mode in both test cases, but not the "page-at-a-time" mode.

File: logging/unit_tests/test_client.py [hunk 2]
```diff
@@ -519,22 +539,40 @@ def test_list_metrics_with_paging(self):
...
 
         # Execute request.
-        metrics, token = client.list_metrics(PAGE_SIZE, TOKEN)
-        # Test values are correct.
+        iterator = client.list_metrics(PAGE_SIZE, TOKEN)
+        metrics = list(iterator)
+        token = iterator.next_page_token
+
+        # First check the token.
+        self.assertIsNone(token)
```
Comment (tseaver) [inline logging/unit_tests/test_client.py]: Likewise, I don't think we care about the token in "list-all-items" mode

File: logging/unit_tests/test_client.py (hunk 1 above)
Comment (dhermes) [inline logging/unit_tests/test_client.py]: I don't have any preferences. What would you like it to do?

File: logging/unit_tests/test_client.py (hunk 2 above)
Comment (dhermes) [inline logging/unit_tests/test_client.py]: Happy to delete this line (and the one that accesses the token).

File: logging/unit_tests/test_client.py (hunk 1 above)
Comment (tseaver) [inline logging/unit_tests/test_client.py]: Perhaps we should adjust `test_list_metrics_with_paging` to read only a page, both passing one token and checking the returned token.  `test_list_metrics_wo_paging` can then ignore all features of paging (page size, tokens, and using the `iterator.pages` attribute).

Comment (dhermes) [inline logging/unit_tests/test_client.py]: Done

**Comment by dhermes** [comment]
@tseaver PTAL

**Pull request merged**